starter/src/main/java/org/teiid/spring/autoconfigure/TeiidServer.java

```diff
@@ -183,7 +184,12 @@ public void addDataSource(VDBMetaData vdb, String sourceBeanName, Object source,
             for (ModelMetaData model : vdb.getModelMetaDatas().values()) {
                 for (SourceMappingMetadata smm : model.getSourceMappings()) {
                     if (smm.getConnectionJndiName() != null && smm.getName().equalsIgnoreCase(sourceBeanName)) {
```

Comment:
Should this check only for a non-null jndi name?  Things like the loopback or a loopback override won't have a source name.

Reply:
The call only gets here on data source instantiation, so it will always have the jndi name. Since the loopback does not have source, it should not get here. So, that raises the question how is the translator added for those cases? I will validate before deploy I guess.